cpp/oneapi/dal/backend/primitives/distance/squared_l2_distance_misc_dpc.cpp

```diff
@@ -70,7 +70,9 @@ sycl::event compute_inner_product(sycl::queue& q,
                                   ndview<Float, 2>& out,
                                   const event_vector& deps) {
     check_inputs(inp1, inp2, out);
-    return gemm(q, inp1, inp2.t(), out, Float(-2.0), Float(+1.0), deps);
+    auto event = gemm(q, inp1, inp2.t(), out, Float(-2.0), Float(+1.0), deps);
+    event.wait_and_throw();
+    return event;
```

Comment:
Do we need to wait here?

Reply:
It didn't work without it before. Will check again.

Reply:
Removed waiting